src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/CertificatePal.cs

```diff
@@ -289,11 +289,11 @@ public string FriendlyName
             {
                 unsafe
                 {
-                    int cbData = 0;
+                    uint cbData = 0;
                     if (!Interop.Crypt32.CertGetCertificateContextPropertyString(_certContext, Interop.Crypt32.CertContextPropId.CERT_FRIENDLY_NAME_PROP_ID, null, ref cbData))
                         return string.Empty;
 
-                    int spanLength = (cbData + 1) / 2;
+                    uint spanLength = (cbData + 1) / 2;
                     Span<char> buffer = spanLength <= 256 ? stackalloc char[spanLength] : new char[spanLength];
                     fixed (char* ptr = &MemoryMarshal.GetReference(buffer))
```

Comment:
This can still overflow. We can compile VSB (`<Compile Include="$(CommonPath)System\Text\ValueStringBuilder.cs" Link="Common\System\Text\ValueStringBuilder.cs" />`) and use it like:

```suggestion
                    uint spanLength = (cbData + 1) / 2;
                    var builder = spanLength <= 256 ?
                        new ValueStringBuilder(stackalloc char[spanLength]) :
                        new ValueStringBuilder(spanLength);
                  
                    fixed (char* ptr = builder)
```

Reply:
I'm not particularly worried about all overflows (I supposed I could just throw a `checked` around everything. My immediate concern is that we never `stackalloc` more than 256 bytes, which containing it to an unsigned int achieves. Is there still a `stack overflow` here that I am missing?

Reply:
To fix the build error in existing code (`CS0266: Cannot implicitly convert type 'uint' to 'int'`), we would need a cast in stackalloc `stackalloc char[(int)spanLength]`.

> Is there still a `stack overflow` here that I am missing?

In the range of `new char[uint.MaxValue - int.MaxValue - 1]` to `new char[uint MaxValue]`, runtime exception is thrown: `System.OverflowException: Arithmetic operation resulted in an overflow`. I think it is alright, when someone has large FriendlyName which exceeds `int.MaxValue`? Otherwise, we can pass a regular StringBuildler:

```c#
#pragma warning disable DLLIMPORTGENANALYZER015 // Use 'GeneratedDllImportAttribute' instead of 'DllImportAttribute' to generate P/Invoke marshalling code at compile time
        // TODO: [DllImportGenerator] Switch to use GeneratedDllImport once we support non-blittable types.

// StringBuilder marshalling overhead is a trade-off to support FriendlyName larger than int.MaxValue
[DllImport("crypt32", EntryPoint = "CertGetCertificateContextProperty", CharSet = CharSet.Unicode, SetLastError = true)]
internal static unsafe extern bool CertGetCertificateContextPropertyString(int pCertContext, int dwPropId, StringBuilder pvData, ref uint pcbData);
#pragma warning restore DLLIMPORTGENANALYZER015

public string FriendlyName
{
    get
    {
        uint cbData = 0;
        if (!Interop.Crypt32.CertGetCertificateContextPropertyString(0, 0, null, ref cbData))
                return string.Empty;

        StringBuilder builder = new();
        if (!Interop.Crypt32.CertGetCertificateContextPropertyString(0, 0, builder, ref cbData) || cbData == 0)
                return string.Empty;

        return builder.ToString();
    }
}
```

Reply:
Ah, right. I forgot that `stackalloc` can't actually take a `uint`.

> runtime exception is thrown

I'm okay with a managed exception here for exceptionally large friendly names (which don't really even make sense).